internal/raft/message/convert.go

```
@@ -0,0 +1,971 @@
+	msgUpdateSetterEquality.Equality.Right = ConvertCommandToMessageExpr(cmd.Right)
+	msgUpdateSetterEquality.Equality.Invert = cmd.Invert
+	return msgUpdateSetterEquality
+}
+
+// ConvertCommandToMessageUpdateSetterRange converts a command.Range to a message.UpdateSetter_Range
+func ConvertCommandToMessageUpdateSetterRange(cmd command.RangeExpr) *UpdateSetter_Range {
+	msgUpdateSetterRange := &UpdateSetter_Range{&RangeExpr{}}
+	msgUpdateSetterRange.Range.Needle = ConvertCommandToMessageExpr(cmd.Needle)
+	msgUpdateSetterRange.Range.Lo = ConvertCommandToMessageExpr(cmd.Lo)
+	msgUpdateSetterRange.Range.Hi = ConvertCommandToMessageExpr(cmd.Hi)
+	msgUpdateSetterRange.Range.Invert = cmd.Invert
+	return msgUpdateSetterRange
+}
+
+// ConvertCommandToMessageUpdateSetter converts a command.UpdateSetter to a message.UpdateSetter
+func ConvertCommandToMessageUpdateSetter(cmd command.UpdateSetter) *UpdateSetter {
+	msgUpdateSetter := &UpdateSetter{}
+	msgUpdateSetter.Cols = cmd.Cols
+	switch cmd.Value.(type) {
+	case command.LiteralExpr:
+		msgUpdateSetter.Value = ConvertCommandToMessageUpdateSetterLiteral(cmd.Value.(command.LiteralExpr))
+	case command.ConstantBooleanExpr:
+		msgUpdateSetter.Value = ConvertCommandToMessageUpdateSetterConstant(cmd.Value.(command.ConstantBooleanExpr))
+	case command.UnaryExpr:
+		msgUpdateSetter.Value = ConvertCommandToMessageUpdateSetterUnary(cmd.Value.(command.UnaryExpr))
+	case command.BinaryExpr:
+		msgUpdateSetter.Value = ConvertCommandToMessageUpdateSetterBinary(cmd.Value.(command.BinaryExpr))
+	case command.FunctionExpr:
+		msgUpdateSetter.Value = ConvertCommandToMessageUpdateSetterFunc(cmd.Value.(command.FunctionExpr))
+	case command.EqualityExpr:
+		msgUpdateSetter.Value = ConvertCommandToMessageUpdateSetterEquality(cmd.Value.(command.EqualityExpr))
+	case command.RangeExpr:
+		msgUpdateSetter.Value = ConvertCommandToMessageUpdateSetterRange(cmd.Value.(command.RangeExpr))
+	}
```

No default value, should probably return an error here